### PR TITLE
Added P2 label to the list of sites in Subscription Manager

### LINKED ISSUE #75078: `/subscriptions/sites`: Add `P2` tag to each P2 site on the list
When the site is a P2 site, it should have the following badge:

Markup on 2023-03-29 at 16:43:23

## Acceptance Criteria

- each P2 site has a badge / label as can be seen in the design: inDLaEQV8jJ21O4WXawIsJ-fi-712-15534

### PR DESCRIPTION
**Pull request by JuanLucha**
⚠️ Please do not merge until the D109848-code patch is merged and deployed. ⚠️ 

Fixes https://github.com/Automattic/wp-calypso/issues/75078

## Proposed Changes

This PR adds the P2 label to the list of sites in the new Subscription Manager, as seen in the Figma file: inDLaEQV8jJ21O4WXawIsJ-fi-712_18319
Screenshot 2023-05-23 at 19 25 24

## Testing Instructions

1. Apply this PR.
2. Go to this file `client/server/pages/index.js` and change the call back in `app.get( [ '/subscriptions', '/subscriptions/*' ], function ( req, res, next ) {` to it ends up like this:
```js
	app.get( [ '/subscriptions', '/subscriptions/*' ], function ( req, res, next ) {
		return next();
	} );
```
3. Log in to your Calypso local and go to `http://calypso.localhost:3000/subscriptions/sites`.
4. If your user has P2 and non-P2 site subscriptions, you should see the P2 tag beside the site's name accordingly.

### CODE REVIEW

**Comment by ivan-ottinger** [comment]
I am reviewing this PR now.

File: client/landing/subscriptions/components/site-list/site-row.tsx
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-list/site-row.tsx]: When we look at the Figma design, there are actually two different tags:

1. blue for `P2` sites
2. green for `Paid` subscriptions

It is not clear to me whether the Paid subscriptions will only have `organization_id` of a certain value, or there will be some other property flagging Paid subscriptions.

With that said, so far it looks like that P2 sites have `organization_id === 1`. Because of that, we might want to check for that value specifically (because simple `organization_id &&` would evaluate as `true` for all `organization_id`s that are "non-zero" - which might include Paid sites as well)

Btw, I noticed that some of the P2s have `organization_id === 2`.

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-list/styles.scss]: Since the same label would be used for Comments list as well, I wonder if it would make sense to move the styling to more general SCSS file that is shared across both lists. We could consider `client/landing/subscriptions/styles.scss` for instance.

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-list/styles.scss]: The `font-size` value is missing `px`, but regardless, we can use variable instead. I think `$font-body-extra-small` should be the right one.

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-list/styles.scss]: I think there still isn't 100% agreement on what color should be used: p1682440802321329-slack-C02TCEHP3HA.

File: client/landing/subscriptions/components/site-list/site-row.tsx
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-list/site-row.tsx]: For `organization_id === 0`, the statement will render `"0"` string instead of `null`:

Markup on 2023-04-26 at 14:30:40

Depending on how the final logic will look like, we may want to use `? <span></span> : null` instead.

**Review by ivan-ottinger — COMMENTED** [review]
The changes display nice blue tag for P2 sites. 🙂 👍🏼 

I have noticed a couple things we will need to address though (please see the comments below).

File: client/landing/subscriptions/components/site-list/site-row.tsx
Comment (JuanLucha) [reply client/landing/subscriptions/components/site-list/site-row.tsx]: Thanks for pointing this out! In fact, making double not (`!! organization_id`) fixes the problem.

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (JuanLucha) [reply client/landing/subscriptions/components/site-list/styles.scss]: We reached a consensus. Applying the correct one now.

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (JuanLucha) [reply client/landing/subscriptions/components/site-list/styles.scss]: Good point! Using the variable now.

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (JuanLucha) [reply client/landing/subscriptions/components/site-list/styles.scss]: Great point! I'll move it right away.

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-list/styles.scss]: I might be missing something, but this import shouldn't be necessary. 🤔

File: client/landing/subscriptions/components/site-list/styles.scss
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-list/styles.scss]: I would personally moved this import to the file where the CSS variable (for the P2 color) is used (`client/landing/subscriptions/styles.scss`) and considered importing just the P2 colors:

```scss
@import "client/assets/stylesheets/_p2-vars";
```

because `client/assets/stylesheets/style` includes a bunch of other imports we don't need for that one P2 color.

**Review by ivan-ottinger — COMMENTED** [review]
Works as expected. 👍🏼👍🏼  Just two more things I noticed. 🙂

File: client/landing/subscriptions/components/site-list/site-row.tsx
Comment (ivan-ottinger) [reply client/landing/subscriptions/components/site-list/site-row.tsx]: Perfect. 👍🏼👍🏼

**Review by ivan-ottinger — APPROVED** [review]
Nice! The proposed changes look good and work as expected 🙂👍🏼👍🏼  (with the D109848-code patch applied to my sandbox - which still needs a review).

- P2 label displays correctly on mobile as well
- I did not notice any regressions, errors, nor warnings.

I am adding the following warning to the PR description - to make sure this doesn't get merged before the patch is in accidentally.

⚠️ Please do not merge until the D109848-code patch is merged and deployed. ⚠️

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8011240

Thank you @JuanLucha for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8011240).

**Pull request merged**
